packages/graphql/src/schema/resolvers/wrapper.ts

```diff
@@ -105,10 +118,45 @@ export const wrapResolver =
             executorConstructorParam.bookmarks = context.driverConfig?.bookmarks;
         }
 
+        if (!neo4jDatabaseInfo?.version) {
+            const dbmsComponentsQuery =
+                "CALL dbms.components() YIELD versions, edition UNWIND versions AS version RETURN version, edition";
+            const dbmsComponentsQueryResult = await new Executor(executorConstructorParam).execute(
+                dbmsComponentsQuery,
+                {},
+                "READ"
+            );
+            const rawRow = dbmsComponentsQueryResult?.records[0];
+            const [rawVersion, edition] = rawRow;
+            neo4jDatabaseInfo = new Neo4jDatabaseInfo(rawVersion as string, edition as Neo4jEdition);
+        }
+        executorConstructorParam.neo4jDatabaseInfo = neo4jDatabaseInfo;
         context.executor = new Executor(executorConstructorParam);
+        context.neo4jDatabaseInfo = neo4jDatabaseInfo;
+
+        return versionMismatchHandler(next)(root, args, context, info);
```

Comment:
It feels slightly strange for the return clause of the `wrapResolver` to be an error handler. Could this not have been caught and then handled, or alternatively, handled within the executor?